app/controllers/doorkeeper/tokens_controller.rb

```diff
@@ -9,29 +9,36 @@ def create
       handle_token_exception e
     end
 
-    # OAuth 2.0 Token Revocation - http://tools.ietf.org/html/rfc7009
-    def revoke
-      # The authorization server, if applicable, first authenticates the client
-      # and checks its ownership of the provided token.
-      #
-      # Doorkeeper does not use the token_type_hint logic described in the
-      # RFC 7009 due to the refresh token implementation that is a field in
-      # the access token model.
-      if authorized?
+     # OAuth 2.0 Token Revocation - https://datatracker.ietf.org/doc/html/rfc7009
+     def revoke
+      # The authorization server responds with HTTP status code 200 if the client
+      # submitted an invalid token or the token has been revoked successfully.
+      if token.blank?
+        render json: {}, status: 200
+      # The authorization server validates [...] and whether the token
+      # was issued to the client making the revocation request. If this
+      # validation fails, the request is refused and the client is informed
+      # of the error by the authorization server as described below.
+      elsif authorized?
         revoke_token
+        render json: {}, status: 200
+      else
+        render json: revocation_error_response, status: :forbidden
```

Comment:
So if I understand it correctly, if not authorized, instead of a quiet "200" pretending that it was successful (which it couldn't have been), it shows an error.